variables.tf

```
@@ -1,5 +1,112 @@
-#variable "example" {
-#  type        = string
-#  default     = ""
-#  description = "This is just an example of a variable"
-#}
+variable "required_apis" {
+  type = map(any)
+  default = {
+    iam             = "iam.googleapis.com"
+    pubsub          = "pubsub.googleapis.com"
+    serviceusage    = "serviceusage.googleapis.com"
+    resourcemanager = "cloudresourcemanager.googleapis.com"
+  }
+}
+
+variable "integration_type" {
+  type        = string
+  default     = "PROJECT"
+  description = "Specify the integration type. Can only be PROJECT or ORGANIZATION. Defaults to PROJECT"
+  validation {
+    condition =  contains(["PROJECT", "ORGANIZATION"], var.integration_type)
+    error_message = "The integration_type must be either PROJECT or ORGANIZATION."
+  }
+}
+
+variable "organization_id" {
+  type        = string
+  default     = ""
+  description = "The organization ID, required if integration_type is set to ORGANIZATION"
+}
+
+variable "project_id" {
+  type        = string
+  default     = ""
+  description = "A project ID different from the default defined inside the provider"
+  validation {
+    condition     = can(regex("(^[a-z][a-z0-9_-]{6,30}[^-]$|^$)", var.project_id))
```

Issue here, the regex has (1char)(6-30char)(1char).  This regex used in go-sdk `(^[a-z][a-z0-9-]{4,28}[a-z0-9]$|^$)`

---

Validation in[ gke module](https://github.com/lacework/terraform-gcp-gke-audit-log/blob/main/variables.tf#L32) and [ audit-log module](https://github.com/lacework/terraform-gcp-audit-log) are different. Should create a task to fix.


Below is the regex Salim added to config and audit-log modules
```suggestion
    condition = can(regex("(^[a-z][a-z0-9-]{4,28}[a-z0-9]$|^$)", var.project_id))
```

---

Yes, have updated this and created a ticket https://lacework.atlassian.net/browse/GROW-1452 for the  [gke](https://github.com/lacework/terraform-gcp-gke-audit-log/blob/main/variables.tf#L32) module to follow the same pattern